src/csvcubed/cli/inspect/metadataprinter.py

```diff
@@ -195,13 +200,29 @@ def generate_codelist_results(self):
         self.result_code_list_cols = select_codelist_cols_by_dataset_url(
             self.csvw_metadata_rdf_graph, self.dataset_url
         )
+        # Retrieving the primary key of the code list to identify the unique identifier
+        result_primary_keys_by_dataset_url: PrimaryKeysByDatasetUrlResult = (
+            select_primary_keys_by_dataset_url(
+                self.csvw_metadata_rdf_graph, self.dataset_url
+            )
+        )
+        primary_keys = result_primary_keys_by_dataset_url.primary_keys
+        
+        # Currently, we do not support composite primary keys.
+        if len(primary_keys) != 1:
+            raise UnsupportedNumOfPrimaryKeysException(
+                num_of_primary_keys=len(primary_keys),
+                table_url=self.dataset_url
+            )        
         (
-            parent_notation_col_name,
-            label_col_name,
-            notation_col_name,
-        ) = self.get_parent_label_notation_col_names(self.result_code_list_cols.columns)
+            parent_notation_col_title,
+            label_col_title,
+            notation_col_title,
+        ) = self.get_parent_label_notation_col_titles(
```

Comment:
Don't forget, this shouldn't be called `notation`/`parent_notation` anymore since the columns don't necessarily represent the notation at all (but there is still notation information in the CSV-W)